### PR TITLE
Add support for passing in parameter instances as streams

### PR DESCRIPTION
**Pull request by philippjfr**
Makes it possible to pass in a Parameter instance as a stream, which will internally be converted to a Params stream. Also adds some validation to ensure the same parameter is not subscribed to multiple times.

- [x] Adds tests

### CODE REVIEW

File: holoviews/streams.py [hunk 1]
```diff
@@ -166,30 +166,41 @@ def trigger(cls, streams):
...
+                pid = id(s.parameterized)
+                if pid in parameterizeds:
+                    overlap = (set(s.parameters) & set(parameterizeds[pid]))
+                    if overlap:
+                        pname = type(s.parameterized).__name__
+                        raise ValueError('Found multiple Params streams '
+                                         'subscribing to the %s parameter(s) '
+                                         'on the %s object. Ensure that '
+                                         'you only subscribe to a parameter '
+                                         'once.' % (list(overlap), pname))
```
Comment (jbednar) [inline holoviews/streams.py]: What would be the effect of simply ignoring the extra subscribers to a parameter?  Would that be safe? If so, could this error simply be skipped?  If not, maybe make a note about why that wouldn't work?

**Comment by jlstevens** [comment]
Looks good. I suppose we don't have any ideas on how to remap parameter names?

One thought I had (probably not a good idea) would be to allow something like:

```
hv.DynamicMap(..., streams={'foo':bar.param.baz})
```

This isn't great as you can't really mix parameter instances and streams nicely. Alternatively:

```
hv.DynamicMap(..., streams=[position_stream, ('foo', bar.param.baz)])
```

But I'm not hot on introducing more tuple formats.

**Comment by philippjfr** [comment]
>Looks good. I suppose we don't have any ideas on how to remap parameter names?

Yeah, it's unfortunate, but I don't want to rush things and come up with something we will hate later so I think this a safe first cut.

File: holoviews/streams.py (hunk 1 above)
Comment (philippjfr) [reply holoviews/streams.py]: I suppose it's safe enough except for the fact that an event may get triggered multiple times. @jlstevens Any opinion here?

**Comment by jlstevens** [comment]
Looks good! Much happier with the new warning.

**Comment by jlstevens** [comment]
Tests are green. Merging.

File: holoviews/streams.py (hunk 1 above)
Comment (jbednar) [reply holoviews/streams.py]: By ignoring I guess I didn't just mean skipping it over, I meant to remove all but one of the subscribers for that parameter.  I guess that's not feasible?

Comment (philippjfr) [reply holoviews/streams.py]: No you'd potentially be modifying user defined objects, which would be weird.

Comment (jbednar) [reply holoviews/streams.py]: Ok, sounds like a warning is best then.

**Pull request merged**